lib/ansible/modules/database/mysql/mysql_info.py

```diff
@@ -0,0 +1,400 @@
+                release=int(release),
+            )
+
+    def __get_master_status(self):
+        res = self.__exec_sql('SHOW MASTER STATUS')
+        if res:
+            for line in res:
+                for vname, val in iteritems(line):
+                    self.info['master_status'][vname] = val
+
+    def __get_slave_status(self):
+        res = self.__exec_sql('SHOW SLAVE STATUS')
+        if res:
+            for line in res:
+                host = line['Master_Host']
+                port = line['Master_Port']
+                user = line['Master_User']
+                self.info['slave_status'][host] = {port: {user: {}}}
+
+                for vname, val in iteritems(line):
+                    if vname not in ('Master_Host', 'Master_Port', 'Master_User'):
+                        self.info['slave_status'][host][port][user][vname] = val
+
+    def __get_slaves(self):
+        res = self.__exec_sql('SHOW SLAVE HOSTS')
+        if res:
+            for line in res:
+                srv_id = line['Server_id']
+                self.info['slave_hosts'][srv_id] = {}
+
+                for vname, val in iteritems(line):
+                    if vname != 'Server_id':
+                        self.info['slave_hosts'][srv_id][vname] = val
+
+    def __get_users(self):
+        res = self.__exec_sql('SELECT * FROM mysql.user')
+        if res:
+            for line in res:
+                host = line['Host']
+                user = line['User']
+                self.info['roles'][host] = {user: {}}
+
+                for vname, val in iteritems(line):
+                    if vname not in ('Host', 'User'):
+                        self.info['roles'][host][user][vname] = val
+
+    def __get_databases(self):
+        query = ('SELECT table_schema AS "name", '
+                 'SUM(data_length + index_length) AS "size" '
+                 'FROM information_schema.TABLES GROUP BY table_schema')
+
+        res = self.__exec_sql(query)
+
+        if res:
+            for db in res:
+                self.info['databases'][db['name']] = {}
+
+                self.info['databases'][db['name']]['size'] = int(db['size'])
+
+    def __exec_sql(self, query, ddl=False):
+        try:
+            self.cursor.execute(query)
+
+            if not ddl:
+                res = self.cursor.fetchall()
+                return res
+            return True
+
+        except Exception as e:
+            self.module.fail_json(msg="Cannot execute SQL '%s': %s" % (query, to_native(e)))
+        return False
+
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = mysql_common_argument_spec()
+    argument_spec.update(
+        login_db=dict(type='str', aliases=['db', 'database']),
+        filter=dict(type='list'),
+    )
+
+    # The module doesn't support check_mode
+    # because of it doesn't change anything
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
```

Comment:
```suggestion
        supports_check_mode=True
```
As suggested by @flowerysong